xformers/benchmarks/LRA/run_tasks.py

```diff
@@ -563,23 +133,158 @@ def get_arg_parser():
         type=dict,
         default=None,
     )
-    parser.add_argument(
-        "--tb_dir",
-        type=str,
-        help="Path to the tensorboard directory",
-        dest="tb_dir",
-        default=f"/checkpoints/{os.getenv('USER')}/xformers/tb",
-    )
     return parser
 
 
+def setup_log(args, attention_name, task) -> Tuple[str, TensorBoardLogger]:
+    experiment_name = f"{task}__{attention_name}"
+    logger = TensorBoardLogger(
+        save_dir=args.checkpoint_dir,
+        name="",  # remove lightning_logs subdirectory
+        version=experiment_name,
+    )
+    log_dir = os.path.join(logger._save_dir, experiment_name)
+    return log_dir, logger
+
+
+def rewrite_hyper(config, rewrites):
+    def replace(config_dict, k, v):
+        if len(k.split(":")) == 1:
+            config_dict[k] = v
+            return
+        first_key = k.split(":")[0]
+        assert first_key in config_dict, first_key
+        k = k[len(first_key) + 1 :]
+        replace(config_dict[first_key], k, v)
+
+    for k, v in rewrites.items():
+        replace(config, k, v)
+    return config
+
+
+def build_dataloaders(
+    args: argparse.Namespace,
+    config_training: Dict,
+    num_workers: int = 4,
+) -> Dict[str, DataLoader]:
+    datasets = {}
+    for component in ("train", "dev", "test"):
+        datasets[component] = LRADataset(
+            file_path=f"datasets/{args.task}.{component}.pickle",
+            seq_len=config_training["seq_len"],
+        )
+
+    # Gradient accumulation
+    accumu_steps = config_training["gradient_accumulation"]
+    logging.info(f"accumu_steps={accumu_steps}")
+
+    # Batch size
+    per_gpu_batch_size = (
+        config_training["batch_size"] // args.world_size // accumu_steps
+    )
+    logging.warning(
+        f"Requested batch size: {config_training['batch_size']}. Given world\
+                size and grad accumulation, per-gpu batch is\
+                {per_gpu_batch_size}"
+    )
+
+    # Training epochs
+    if accumu_steps > 1:
+        config_training["num_train_steps"] *= accumu_steps
```

Comment:
I'm not sure that this is still required with lightning, it handles grad accumulation out of the box, right ?

Reply:
I thought so too but based off [this warning](https://pytorch-lightning.readthedocs.io/en/latest/advanced/training_tricks.html#accumulate-gradients), it sounds like gradient accumulation coupled with DDP behaves differently. I'll look into it in more detail.

Reply:
this is fine, it just means that prior to the optimizer step the gradients will not be in sync across the fleet, while it's the case without the accumulation. It's a useful warning if you were to peek into the gradients on a per-rank basis, and decide on something from that, in that case triggering the grad acc could mess with your logic. We're not doing that here, simply training over all the ranks, so the default lightning behaviour should be fine

Reply:
Ah makes sense—thanks! Will get rid of this block then.